### DISCUSSION TITLE
Advantage Async version

### DISCUSSION THREAD

**Discussion by Tockra** [dani-garcia/vaultwarden#1717] [General]
Hi Guys,

is there any reason why you have the async branch ? 
Are there any advantages to use/build the async version instead of the main branch?

I know rust crates which has async versions to allow users the using of the crate in an async context, but your project is a whole project. So I don't understand the reason to separate it into two branches (with async and without async).

T

**Comment by RealOrangeOne** [comment]
As far as I know, it's in a separate branch because it's in progress, to be merged in future. I don't think there's an intention to maintain both at the same time, because like you say that doesn't make much sense for an application

**Comment by BlackDex** [comment]
Well using async crates mean to also have some changes done on other part of the code.
It isn't as simple as just switching some feature flags.

Also, we are waiting mostly on the rocket crate to finalize on v0.5 which should support async.

To not have to do a very very big rebase on the project @dani-garcia is having a separate branch where he (and others) can test this async version and keep it up-to-date with the changes to master.

The end-goal is to have the async branch as default if all works.
